versioning/BRM/extentmap.cpp

```diff
@@ -219,15 +219,27 @@ boost::mutex ExtentMapRBTreeImpl::fInstanceMutex;
 ExtentMapRBTreeImpl* ExtentMapRBTreeImpl::fInstance = nullptr;
 
 /*static*/
-ExtentMapRBTreeImpl* ExtentMapRBTreeImpl::makeExtentMapRBTreeImpl(unsigned key, off_t size, bool readOnly)
+ExtentMapRBTreeImpl* ExtentMapRBTreeImpl::makeExtentMapRBTreeImpl(unsigned key, off_t size, bool readOnly,
+                                                                  bool& emLocked,
+                                                                  const MasterSegmentTable* emSegTable)
 {
   boost::mutex::scoped_lock lk(fInstanceMutex);
 
   if (fInstance)
   {
     if (key != fInstance->fManagedShm.key())
     {
+      if (emSegTable)
```

Comment:
@drrtuy I have more conservative idea, how about to grab "write lock" inside grabEmEntyTable
right after this condition 
 `if (!fPExtMapRBTreeImpl || fPExtMapRBTreeImpl->key() != (uint32_t)fEMRBTreeShminfo->tableShmkey)`
 
in this case have a "write lock" either EM is nullptr for current process or the keys are different between current process and global segment table. What do you think?

Reply:
Yes, but this case only could be possible when fPExtMapRBTreeImpl == nullptr, and this only could possible when the process starts (PP, DML etc) so only once.